store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -300,8 +300,50 @@ public static void parseModifyVolumeRequest(ModifyVolumeRequest req) throws Serv
         else if (vol.getStoreType().equals(Volume.StoreType.EXTERNAL)) {
             if (volInfo.getName() != null) {
                 builder.setName(volInfo.getName());
+                String extRootPath = "/" + readStorageType(req.getId()) + "-" + volInfo.getName() + "-" + readGlobalBucketConfigId(req.getId());
+                builder.setPath(extRootPath, false);
             }
         }
         mgr.update(builder.build());
     }
+
+    /**
```

Comment:
I think you don't need to create the below methods, you can change this code as shown below(instead of creating an object of `ExternalVolumeInfoHandler` multiple times )
```
ExternalVolumeInfoHandler extVolInfoHandler = new ExternalVolumeInfoHandler(Provisioning.getInstance());
JSONObject jsonProperties = extVolInfoHandler.readServerProperties(id);
String  storageType = properties.getString(AdminConstants.A_VOLUME_STORAGE_TYPE);
String globalBucketConfigId = properties.getString(AdminConstants.A_VOLUME_GLB_BUCKET_CONFIG_ID);
```

Reply:
Fixed